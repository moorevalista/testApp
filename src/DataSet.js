const DataSet = [
  {
    id_user: 1,
    first_name: 'John',
    last_name: 'Lennon',
    favorite: false,
    cats: [
      {
        cat_name: 'Snowball',
        cat_age: '4 years 0 month'
      },
      {
        cat_name: 'Blackbly',
        cat_age: '2 years 3 months'
      },
    ]
  },

  {
    id_user: 2,
    first_name: 'Paul',
    last_name: 'McCartney',
    favorite: false,
    cats: [
      {
        cat_name: 'Oscar',
        cat_age: '3 years 4 months'
      },
    ]
  },

  {
    id_user: 3,
    first_name: 'Georgia',
    last_name: 'Harrison',
    favorite: false,
    cats: [
      {
        cat_name: 'Bono',
        cat_age: '1 years 6 months'
      },
      {
        cat_name: 'Chiko',
        cat_age: '3 years 1 months'
      },
      {
        cat_name: 'Thomas',
        cat_age: '2 years 9 months'
      },
    ]
  },

  {
    id_user: 4,
    first_name: 'Ringo',
    last_name: 'Starr',
    favorite: false,
    cats: [
      {
        cat_name: 'Snow White',
        cat_age: '5 years 2 months'
      },
      {
        cat_name: 'Adam',
        cat_age: '4 years 4 months'
      },
    ]
  },

  {
    id_user: 5,
    first_name: 'Andi',
    last_name: 'Purwanto',
    favorite: false,
    cats: [
      {
        cat_name: 'Snowy',
        cat_age: '3 years 11 months'
      },
      {
        cat_name: 'Bocil',
        cat_age: '4 years 9 months'
      },
    ]
  },

  {
    id_user: 6,
    first_name: 'Kevin',
    last_name: 'Bastian',
    favorite: false,
    cats: [
      {
        cat_name: 'Blacky',
        cat_age: '6 years 3 months'
      },
      {
        cat_name: 'Kitty',
        cat_age: '5 years 5 months'
      },
      {
        cat_name: 'Kuncoro',
        cat_age: '4 years 4 months'
      },
      {
        cat_name: 'Kucrit',
        cat_age: '2 years 2 months'
      },
    ]
  },

  {
    id_user: 7,
    first_name: 'Rayyan',
    last_name: 'Moore',
    favorite: false,
    cats: [
      {
        cat_name: 'Choco',
        cat_age: '1 years 2 months'
      },
      {
        cat_name: 'Chippy',
        cat_age: '10 years 3 months'
      },
    ]
  },

  {
    id_user: 8,
    first_name: 'Onno',
    last_name: 'Soppo',
    favorite: false,
    cats: [
      {
        cat_name: 'Bocil',
        cat_age: '3 years 0 month'
      },
      {
        cat_name: 'Belang',
        cat_age: '6 years 7 months'
      },
      {
        cat_name: 'Ibas',
        cat_age: '3 years 7 months'
      },
    ]
  },

  {
    id_user: 9,
    first_name: 'Soppo',
    last_name: 'Jarwo',
    favorite: false,
    cats: [
      {
        cat_name: 'Entus',
        cat_age: '8 years 10 months'
      },
      {
        cat_name: 'Entis',
        cat_age: '4 years 4 months'
      },
    ]
  },

  {
    id_user: 10,
    first_name: 'Dodi',
    last_name: 'Setiawan',
    favorite: false,
    cats: [
      {
        cat_name: 'Mari',
        cat_age: '7 years 2 months'
      },
      {
        cat_name: 'Maru',
        cat_age: '6 years 10 months'
      },
    ]
  },
  {
    id_user: 11,
    first_name: 'Pangeran',
    last_name: 'Sentosa',
    favorite: false,
    cats: [
      {
        cat_name: 'Maro',
        cat_age: '3 years 10 month'
      },
      {
        cat_name: 'Maroon',
        cat_age: '3 years 3 months'
      },
      {
        cat_name: 'Soyo',
        cat_age: '2 years 9 months'
      },
    ]
  },

  {
    id_user: 12,
    first_name: 'Hendra',
    last_name: 'Saputra',
    favorite: false,
    cats: [
      {
        cat_name: 'Oncel',
        cat_age: '7 years 4 months'
      },
    ]
  },

  {
    id_user: 13,
    first_name: 'Rico',
    last_name: 'Kurnia',
    favorite: false,
    cats: [
      {
        cat_name: 'Bulu',
        cat_age: '2 years 6 months'
      },
      {
        cat_name: 'Soju',
        cat_age: '3 years 6 months'
      },
      {
        cat_name: 'Buhan',
        cat_age: '3 years 9 months'
      },
    ]
  },

  {
    id_user: 14,
    first_name: 'Benny',
    last_name: 'Sianturi',
    favorite: false,
    cats: [
      {
        cat_name: 'Smokey',
        cat_age: '5 years 4 months'
      },
      {
        cat_name: 'Aden',
        cat_age: '3 years 4 months'
      },
    ]
  },

  {
    id_user: 15,
    first_name: 'Amran',
    last_name: 'Prabowo',
    favorite: false,
    cats: [
      {
        cat_name: 'Unyil',
        cat_age: '3 years 1 months'
      },
      {
        cat_name: 'Unyu',
        cat_age: '4 years 2 months'
      },
    ]
  },

  {
    id_user: 16,
    first_name: 'Reza',
    last_name: 'Arep',
    favorite: false,
    cats: [
      {
        cat_name: 'Codot',
        cat_age: '1 years 3 months'
      },
      {
        cat_name: 'Comel',
        cat_age: '4 years 1 months'
      },
      {
        cat_name: 'Cimol',
        cat_age: '4 years 7 months'
      },
      {
        cat_name: 'Codet',
        cat_age: '1 years 2 months'
      },
    ]
  },

  {
    id_user: 17,
    first_name: 'Deni',
    last_name: 'Sauron',
    favorite: false,
    cats: [
      {
        cat_name: 'Chece',
        cat_age: '2 years 0 month'
      },
      {
        cat_name: 'Chappy',
        cat_age: '3 years 3 months'
      },
    ]
  },

  {
    id_user: 18,
    first_name: 'Adi',
    last_name: 'Wijaya',
    favorite: false,
    cats: [
      {
        cat_name: 'Nino',
        cat_age: '1 years 0 month'
      },
      {
        cat_name: 'Nunu',
        cat_age: '3 years 4 months'
      },
      {
        cat_name: 'Nuno',
        cat_age: '3 years 2 months'
      },
    ]
  },

  {
    id_user: 19,
    first_name: 'Afrizal',
    last_name: 'Situbondo',
    favorite: false,
    cats: [
      {
        cat_name: 'Ello',
        cat_age: '2 years 10 months'
      },
      {
        cat_name: 'Ella',
        cat_age: '3 years 1 months'
      },
    ]
  },

  {
    id_user: 20,
    first_name: 'Yunus',
    last_name: 'Susanto',
    favorite: false,
    cats: [
      {
        cat_name: 'Momo',
        cat_age: '4 years 2 months'
      },
      {
        cat_name: 'Mimi',
        cat_age: '2 years 2 months'
      },
    ]
  },
];

export default DataSet;