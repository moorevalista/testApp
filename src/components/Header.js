import React from "react"
import { StyleSheet, Text, View } from "react-native"

export default function Header(props) {
  const data = props.data.filter(item => item.id_user === props.master);
  
  return (
    data.length > 0 ?
    <View style={styles.GroupHeader}>
      <View style={styles.GroupLeft}>
        <View style={styles.Avatar}>
          <Text style={styles.InitialLabel}>{data[0].first_name.substring(0,1) + data[0].last_name.substring(0,1)}</Text>
        </View>
      </View>
      <View style={styles.GroupName}>
        <Text style={styles.LabelName}>Master: {data[0].first_name + ' ' + data[0].last_name}</Text>
      </View>
    </View>
    :
    <View style={styles.GroupHeader}>
      <View style={styles.GroupLeft}>
        
      </View>
    </View>
  )
}

const styles = StyleSheet.create({
  GroupHeader: {
    position: 'relative',
    display: "flex",
    flexDirection: "row",
    alignItems: "center",
    paddingHorizontal: '5%',
    paddingVertical: '2%',
    // marginBottom: '5%',
    justifyContent: 'center',
  },
  GroupLeft: {
    marginRight: '5%',
    width: 40,
    height: 40,
  },
  Avatar: {
    display: "flex",
    flexDirection: "column",
    justifyContent: "center",
    alignItems: "center",
    borderRadius: 20,
    backgroundColor: "white",
    /*  linear-gradient(0deg, rgba(255, 255, 255, 1), rgba(255, 255, 255, 1)), linear-gradient(0deg, rgba(251,231,198,0.9) 0%, rgba(251,231,198,0.6) 100%, )  */
    borderWidth: 1,
    borderStyle: "solid",
    borderColor: "rgba(54,163,136,1)",
    backdropFilter: "blur(12px)",
    width: 40,
    height: 40,
  },
  InitialLabel: {
    fontSize: 15,
    fontFamily: "SFProText-Regular",
    fontWeight: "600",
    letterSpacing: 0.3,
    color: "rgba(54,163,136,1)",
    textAlign: "center",
    justifyContent: "center",
  },

  GroupName: {
    // borderWidth: 2
  },
  LabelName: {
    fontSize: 14,
    fontFamily: "CircularStd-Medium",
    fontWeight: "500",
    letterSpacing: 0.28,
    color: "rgba(64,64,64,1)",
  },
})
