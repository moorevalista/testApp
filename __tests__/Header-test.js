import React from 'react';
import renderer from 'react-test-renderer';
import Header from '../src/components/Header';
import Adapter from 'enzyme-adapter-react-16';
import {shallow, configure} from 'enzyme';
import DataSet from '../src/DataSet';

jest.mock('react-native/Libraries/Animated/NativeAnimatedHelper');

configure({adapter: new Adapter(), disableLifecycleMethods: true})

// const appWrapper = shallow(<BoxUser />)

const data = DataSet;

describe('BoxUser Component', () => {
  
  it('should renders correctly', () => {
    expect(renderer.create(<Header master={1} data={data} />).toJSON()).toMatchSnapshot();
  })

})