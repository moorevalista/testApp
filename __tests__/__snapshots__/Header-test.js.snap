// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BoxUser Component should renders correctly 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "display": "flex",
      "flexDirection": "row",
      "justifyContent": "center",
      "paddingHorizontal": "5%",
      "paddingVertical": "2%",
      "position": "relative",
    }
  }
>
  <View
    style={
      Object {
        "height": 40,
        "marginRight": "5%",
        "width": 40,
      }
    }
  >
    <View
      style={
        Object {
          "alignItems": "center",
          "backdropFilter": "blur(12px)",
          "backgroundColor": "white",
          "borderColor": "rgba(54,163,136,1)",
          "borderRadius": 20,
          "borderStyle": "solid",
          "borderWidth": 1,
          "display": "flex",
          "flexDirection": "column",
          "height": 40,
          "justifyContent": "center",
          "width": 40,
        }
      }
    >
      <Text
        style={
          Object {
            "color": "rgba(54,163,136,1)",
            "fontFamily": "SFProText-Regular",
            "fontSize": 15,
            "fontWeight": "600",
            "justifyContent": "center",
            "letterSpacing": 0.3,
            "textAlign": "center",
          }
        }
      >
        JL
      </Text>
    </View>
  </View>
  <View
    style={Object {}}
  >
    <Text
      style={
        Object {
          "color": "rgba(64,64,64,1)",
          "fontFamily": "CircularStd-Medium",
          "fontSize": 14,
          "fontWeight": "500",
          "letterSpacing": 0.28,
        }
      }
    >
      Master: 
      John Lennon
    </Text>
  </View>
</View>
`;
